### PR TITLE
Fix a couple of issue with move of nested directories

### PR DESCRIPTION
**Pull request by ogoffart**
Fix issue #6694 and some other move problems.

### CODE REVIEW

File: src/libsync/discovery.cpp [hunk 1]
```diff
@@ -496,8 +496,8 @@ void ProcessDirectoryJob::processFileAnalyzeRemoteInfo(
         bool wasDeletedOnServer = _discoveryData->findAndCancelDeletedJob(originalPath).first;
 
         auto postProcessRename = [this, item, base, originalPath](PathTuple &path) {
-            auto adjustedOriginalPath = _discoveryData->adjustRenamedPath(originalPath);
-            _discoveryData->_renamedItems.insert(originalPath, path._target);
+        auto adjustedOriginalPath = _discoveryData->adjustRenamedPath(originalPath, SyncFileItem::Up);
+            _discoveryData->_renamedItemsRemote.insert(originalPath, path._target);
```
Comment (ckamm) [inline src/libsync/discovery.cpp]: odd indent

File: src/libsync/owncloudpropagator.h [hunk 2]
```diff
@@ -506,6 +506,10 @@ class OwncloudPropagator : public QObject
     bool createConflict(const SyncFileItemPtr &item,
         PropagatorCompositeJob *composite, QString *error);
 
+
+    QMap<QString, QString> _renamedDirectories;
```
Comment (ckamm) [inline src/libsync/owncloudpropagator.h]: Will not splitting this into local and remote renames lead to bugs similar to in the discovery? 

Please add a explanatory comment too: Why is this necessary when `adjustRenamedPath()` is already called on _item->_file during discovery?

File: test/testsyncmove.cpp [hunk 3]
```diff
@@ -660,7 +659,57 @@ private slots:
...
+        modifier.rename("FolA", "FolA_Renamed");
+        modifier.rename("FolA_Renamed/FolB", "FolB_Renamed");
+        modifier.rename("FolB_Renamed/FolC", "FolA");
+        modifier.rename("FolA/FolD", "FolA/FolD_Renamed");
+        modifier.mkdir("FolB_Renamed/New");
+        modifier.rename("FolA/FolD_Renamed/FolE", "FolB_Renamed/New/FolE");
+        auto expected = local ? fakeFolder.currentLocalState() : fakeFolder.currentRemoteState();
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), expected);
+        QCOMPARE(fakeFolder.currentRemoteState(), expected);
```
Comment (ckamm) [inline test/testsyncmove.cpp]: Do we already have a test for interacting renames on different sides?
```
local: A -> B
remote: A/foo -> A/bar
```
and the other way around.

File: src/libsync/owncloudpropagator.h (hunk 2 above)
Comment (ogoffart) [reply src/libsync/owncloudpropagator.h]: > Will not splitting this into local and remote renames lead to bugs similar to in the discovery?

No, because in the discovery we map  original -> current local or server path, which might be different from target.

But here we always map origin->target and that is the same for both local and server.

I'm adding a comment

File: test/testsyncmove.cpp (hunk 3 above)
Comment (ogoffart) [reply test/testsyncmove.cpp]: That's right,  but that'd be another issue.

Comment (ogoffart) [reply test/testsyncmove.cpp]: Fixed now

Comment (ckamm) [reply test/testsyncmove.cpp]: Great, looks good

**Comment by ogoffart** [comment]
@ckamm  ok to merge?
